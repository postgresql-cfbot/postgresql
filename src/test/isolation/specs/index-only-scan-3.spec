# A variant of index-only-scan-2.spec that is a true conflict, detected via
# heap tuple locks only.

setup
{
  CREATE TABLE account (id1 int, id2 int, balance int);
  CREATE UNIQUE INDEX ON account(id1);
  CREATE UNIQUE INDEX ON account(id2);
  INSERT INTO account VALUES (1, 1, 100), (2, 2, 100);
}
setup
{
  VACUUM account;
}

teardown
{
  DROP TABLE account;
}

session "s1"
setup { BEGIN ISOLATION LEVEL SERIALIZABLE; SET enable_seqscan = off; }
step "r1" { SELECT id1 FROM account WHERE id1 = 2; }
step "w1" { UPDATE account SET balance = 200 WHERE id1 = 1; }
step "c1" { COMMIT; }

session "s2"
setup { BEGIN ISOLATION LEVEL SERIALIZABLE; SET enable_seqscan = off; }
step "r2" { SELECT id2 FROM account WHERE id2 = 1; }
step "w2" { UPDATE account SET balance = 200 WHERE id2 = 2; }
step "c2" { COMMIT; }

permutation "r1" "r2" "w1" "w2" "c1" "c2"
