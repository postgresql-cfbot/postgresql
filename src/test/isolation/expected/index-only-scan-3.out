Parsed test spec with 2 sessions

starting permutation: r1 r2 w1 w2 c1 c2
step r1: SELECT id1 FROM account WHERE id1 = 2;
id1            

2              
step r2: SELECT id2 FROM account WHERE id2 = 1;
id2            

1              
step w1: UPDATE account SET balance = 200 WHERE id1 = 1;
step w2: UPDATE account SET balance = 200 WHERE id2 = 2;
step c1: COMMIT;
step c2: COMMIT;
ERROR:  could not serialize access due to read/write dependencies among transactions
